### PR TITLE
[ONNX] Export KLDivLoss

### PR DESCRIPTION
**Pull request by neginraoof**
Enable export for KLDivLoss

### CODE REVIEW

File: torch/onnx/symbolic_opset9.py [hunk 1]
```diff
@@ -2423,3 +2423,37 @@ def take(g, self, index):
...
+    output = mul(g, exp_, diff_)
+    return output
+
+
+def _kl_div_non_log_target(g, input, target):
+    log_ = log(g, target)
+    diff_ = sub(g, log_, input)
+    output_pos = mul(g, target, diff_)
+    zeros_ = zeros_like(g, output_pos)
+    mask_ = gt(g, target, g.op("Constant", value_t=torch.tensor(0)))
```
Comment (spandantiwari) [inline torch/onnx/symbolic_opset9.py:2440]: Just curious - in the `gt` op we are checking against a `onnx::Constant` which contains 0. The type seems to be int. Do we need to do something to handle situations when `target` is a some other type, say float? onnx::Greater requires the two inputs to be of the same type.

File: torch/onnx/symbolic_opset9.py [hunk 2]
```diff
@@ -2423,3 +2423,37 @@ def take(g, self, index):
     out = index_select(g, self_flattened, 0, index)
     out = reshape_as(g, out, index)
     return out
+
+
+def _kl_div_log_target(g, input, target):
```
Comment (spandantiwari) [inline torch/onnx/symbolic_opset9.py:2428]: Are `_kl_div_log_target` and `_kl_div_non_log_target` names of the symbolics coming from aten itself or these just helper functions? If it is the latter I would suggest naming them differently, possibly with the word `_impl` in the name, to distinguish them from real symbolic entry points in this file. E.g. `_kl_div_log_target_impl` and `_kl_div_non_log_target_impl`.

File: torch/onnx/symbolic_opset9.py [hunk 3]
```diff
@@ -2423,3 +2423,37 @@ def take(g, self, index):
...
+        output = _kl_div_non_log_target(g, input, target)
+
+    if reduction == 0:
+        return output
+    elif reduction == 1:
+        return g.op("ReduceMean", output, keepdims_i=0)
+    elif reduction == 2:
+        return g.op("ReduceSum", output, keepdims_i=0)
+    else:
+        return _unimplemented("kl_div", "reduction other than none, mean, or sum.")
```
Comment (spandantiwari) [inline torch/onnx/symbolic_opset9.py:2459]: nit: maybe phrase the error message as a more complete sentence. Anything actionable we can say in the message?

File: torch/onnx/symbolic_opset9.py (hunk 1 above)
Comment (neginraoof) [reply torch/onnx/symbolic_opset9.py:2440]: Yes, this is going to be handled by the scalar type analysis in PR: https://github.com/pytorch/pytorch/pull/37787

Comment (spandantiwari) [reply torch/onnx/symbolic_opset9.py:2440]: Great.

**Review by spandantiwari — APPROVED** [review]
LGTM. Can we update the helper method names from _kl_div_log_target to _kl_div_log_target_impl?

**Comment by neginraoof** [comment]
@Spa

> LGTM. Can we update the helper method names from _kl_div_log_target to _kl_div_log_target_impl?

Yes, this is done already.

**Review by bzinodev — APPROVED** [review]

**Comment by neginraoof** [comment]
@bzinodev  Is there any related issue with tests or anything blocking merge?

**Comment by neginraoof** [comment]
@bzinodev Can we merge this PR?

**Comment by neginraoof** [comment]
@bzinodev CI is green. Would you please help merge this PR?

**Comment by neginraoof** [comment]
@bzinodev 
CI is green. Would you please help merge this PR? Thanks!

**Pull request closed without merging**